Fix spammy forum notifications.

custode opened the pull request:
Separate the subscription of forums from topics. E.g. Topic's post updates will no longer send email notifications to the forum subscriber. Forum subscriber will now only receive new topic notifications.

Also updated the title of the subscribe/unsubscribe button to specify what the user are subscribing to.

Lastly, added the missing GET link to allow user to unsubscribe immediately from the link in email.

allenwq commented:
@custode You should remove the `post` then. 

Also, we should use `link_to` in the email template, currently it's a raw link.

custode commented:
@allenwq The `post` is currently used by the forum button.

Updated the email template to make use of link_to and replaced the unsubscribe message.

allenwq commented:
@custode OK, think leave them then.
